### PR TITLE
Ndlano/issues#1081 support attachment nodes

### PR DESCRIPTION
**Pull request by sebastianjg**
Resolves https://github.com/NDLANO/Issues/issues/1081
Currently supports editing of file title metadata

Test article: `/subject-matter/learning-resource/4835/edit/nb`

Test procedure:
- [ ] Check that file list component mounts when page loads
- [ ] Check that file list components enters edit mode on click
- [ ] Check that fields for title editing changes file titles and keeps state when entering / exiting edit mode
- [ ] Check that file nodes and title metadata changes are saved on article save

### CODE REVIEW

File: src/components/SlateEditor/plugins/file/EditFile.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            <section {...classes()}>
+              <h1 {...classes('heading')}>{heading}</h1>
+              <ul {...classes('files')}>
+                {files.map(file => [
+                  <File
+                    key={`file-${file.id}-${file.title}`}
+                    file={file}
+                    id="file-embed"
+                  />,
+                  <SlateInputField
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/file/EditFile.jsx]: Mangler ikke key her?

File: src/components/SlateEditor/plugins/file/Filelist.jsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    this.setState(prevState => ({ editMode: !prevState.editMode }));
+  }
+
+  render() {
+    const { files } = this.state;
+    const { t } = this.props;
+
+    return (
+      <Fragment>
+        {this.state.editMode ? (
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/file/Filelist.jsx]: hente den under destruct?

File: src/components/SlateEditor/plugins/file/Filelist.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            heading={t(`form.file.label`)}
+            files={files}
+            onExit={this.toggleEdit}
+            onFileListInputChange={this.onFileInputChange}
+            submitted={false}
+          />
+        ) : (
+          <div
+            role="button"
+            style={{ cursor: 'pointer' }}
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/file/Filelist.jsx]: inline css?

File: src/components/SlateEditor/plugins/file/EditFile.jsx (hunk 1 above)
Comment (sebastianjg) [reply src/components/SlateEditor/plugins/file/EditFile.jsx]: Lagt til under

Comment (chrpeter) [reply src/components/SlateEditor/plugins/file/EditFile.jsx]: Det er vel fordi du bruker title som kan endre seg som en del av key'en, dermed er ikke key likt og det kan få noen uheldige sidekonsekvenser?

**Comment by sebastianjg** [comment]
CSS er rettet

**Pull request merged**